cmd/protecodeExecuteScan_test.go

```diff
@@ -404,6 +404,16 @@ func TestGetTarName(t *testing.T) {
 			"",
 			"abc.tar",
 		},
+		"with sha ": {
```

Comment:
No need for a trailing space in my opinion. Also remove it from `"without version "` I would say. Looks like a typo to me.

Reply:
done as suggested. 
I guess without version is also a sinful test ? If user doesn't specify anything.
We can remove it as well.

Reply:
I don't understand the comment. (Did you really mean "sinful" or are you talking German: "Sinnvoll"?)

Reply:
Sorry about the misspell above.

Reply:
> I guess without version is also a sensible test ? If user doesn't specify anything.


I'm not sure I understand the question. It is not related to the trailing spaces, right? I don't think it is necessary to remove an existing test (`"without version"`). I do believe it is good to test the sha digest without a version. I am not so sure about testing it with a version, as it does not make sense, as you will not be able to download a docker image with sha digest and version.

Reply:
> > I guess without version is also a sensible test ? If user doesn't specify anything.
> 
> I'm not sure I understand the question. It is not related to the trailing spaces, right? I don't think it is necessary to remove an existing test (`"without version"`). I do believe it is good to test the sha digest without a version. I am not so sure about testing it with a version, as it does not make sense, as you will not be able to download a docker image with sha digest and version.

I guess you can't download a dockerImage with **version**. User can download it with **shadigest** or providing the dockerImage with **tag**.
artifactVersion has no significance or meaning here - just a string that appends to the dockerImage. So that user can easily detect it at protecode backend. 

https://sap.github.io/jenkins-library/steps/protecodeExecuteScan/#artifactversion